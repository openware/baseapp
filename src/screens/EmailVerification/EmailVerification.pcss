.pg-emailverification-container {
  align-items: center;
  background: url(/static/media/background.a12e192e.svg);
  background-size: cover;
  display: flex;
  height: calc(100vh - calc(var(--gap) * 9.84));
  justify-content: center;
  width: 100%;

  .pg-emailverification {
    align-items: center;
    background: var(--body-background-color);
    border-radius: calc(var(--big-gap) * 0.05);
    box-shadow: 0 10px 15px var(--shadow-color-level-2);
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: calc(var(--big-gap) * 7.07);

    &-title {
      align-items: center;
      background: var(--subheader-background-color);
      border-bottom: 2px solid var(--divider-color-level-2);
      color: var(--primary-text-color);
      display: flex;
      font-size: calc(var(--big-gap) * 0.23);
      height: var(--big-gap);
      justify-content: center;
      text-transform: uppercase;
      width: 100%;
    }

    &-body {
      background: var(--body-background-color);
      display: flex;
      flex-direction: column;
      height: calc(var(--big-gap) * 3.6);
      justify-content: space-around;

      &-text {
        align-items: center;
        color: var(--secondary-contrast-cta-color);
        display: flex;
        font-size: calc(var(--big-gap) * 0.27);
        justify-content: center;
        padding: 0 calc(var(--big-gap) * 0.9);
        text-align: center;
      }

      &-container {
        align-items: center;
        display: flex;
        justify-content: center;

        &-button {
          background: none;
          border: none;
          color: var(--secondary-contrast-cta-color);
          cursor: pointer;
          font-size: calc(var(--big-gap) * 0.23);
          outline: none;
          text-decoration: underline;
          transition: 0.2s;

          &:hover {
            color: var(--primary-text-color);
            transition: 0.2s;
          }
        }
      }
    }
  }
}
