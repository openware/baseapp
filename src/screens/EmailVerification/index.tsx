import { History } from 'history';
import * as React from 'react';
import {Button, Spinner} from 'react-bootstrap';
import {
    injectIntl,
} from 'react-intl';
import { connect, MapStateToProps } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { compose } from 'redux';
import { captchaType } from '../../api/config';
import { IntlProps } from '../../';
import { Captcha } from '../../components';
import { EMAIL_REGEX, setDocumentTitle } from '../../helpers';
import {
    emailVerificationFetch,
    GeetestCaptchaResponse,
    resetCaptchaState,
    RootState,
    selectCaptchaResponse,
    selectCurrentLanguage,
    selectGeetestCaptchaSuccess,
    selectMobileDeviceState,
    selectRecaptchaSuccess,
    selectSendEmailVerificationError,
    selectSendEmailVerificationLoading,
    selectSendEmailVerificationSuccess,
    selectUserInfo,
    User,
} from '../../modules';
import { CommonError } from '../../modules/types';

interface OwnProps {
    history: History;
    location: {
        state: {
            email: string;
        };
    };
    success: boolean;
    error?: CommonError;
}

interface DispatchProps {
    emailVerificationFetch: typeof emailVerificationFetch;
    resetCaptchaState: typeof resetCaptchaState;
}

interface ReduxProps {
    emailVerificationLoading: boolean;
    isMobileDevice: boolean;
    captcha_response?: string | GeetestCaptchaResponse;
    reCaptchaSuccess: boolean;
    geetestCaptchaSuccess: boolean;
    user: User;
}

type Props = DispatchProps & ReduxProps & OwnProps & IntlProps;

class EmailVerificationComponent extends React.Component<Props> {
    public componentDidMount() {
        setDocumentTitle('Email verification');

        if (!this.props.location.state) {
            this.props.history.push('/signin');
        }
    }

    public translate = (id: string) => this.props.intl.formatMessage({ id });

    public renderCaptcha = () => {
        const { error, success } = this.props;

        return (
            <Captcha
                error={error}
                success={success}
            />
        );
    };

    public render() {
        const { emailVerificationLoading, isMobileDevice } = this.props;

        const title = this.props.intl.formatMessage({ id: 'page.header.signUp.modal.header' });
        const text = this.props.intl.formatMessage({ id: 'page.header.signUp.modal.body' });
        const button = this.props.intl.formatMessage({ id: 'page.resendConfirmation' });

        return (
            <div className="pg-emailverification-container">
                <div className="pg-emailverification">
                    {!isMobileDevice && <div className="pg-emailverification-title">{title}</div>}
                    <div className="pg-emailverification-body">
                        <div className="pg-emailverification-body-text">{text}</div>
                        {this.renderCaptcha()}
                        {
                            !isMobileDevice && (
                                <div className="pg-emailverification-body-container">
                                    {emailVerificationLoading ? <Spinner animation="border" variant="primary"/> :
                                        <button className="pg-emailverification-body-container-button"
                                                onClick={this.handleClick}
                                                disabled={this.disableButton()}>{button}</button>}
                                </div>)
                        }
                        {isMobileDevice &&
                            (<div className="pg-emailverification-body-container">
                              <Button
                                block={true}
                                type="button"
                                onClick={this.handleClick}
                                size="lg"
                                variant="primary"
                              >
                                  {this.props.intl.formatMessage({ id:  'page.mobile.reset.header' })}
                              </Button>
                            </div>)
                        }
                    </div>
                </div>
            </div>
        );
    }

    private handleClick = () => {
        const { captcha_response } = this.props;

        switch (captchaType()) {
            case 'recaptcha':
            case 'geetest':
                this.props.emailVerificationFetch({
                    email: this.props.location.state.email,
                    captcha_response,
                });
                break;
            default:
                this.props.emailVerificationFetch({
                    email: this.props.location.state.email,
                });
                break;
        }

        this.props.resetCaptchaState();
    };

    private disableButton = (): boolean => {
        const {
            location,
            geetestCaptchaSuccess,
            reCaptchaSuccess,
        } = this.props;
        const captchaTypeValue = captchaType();

        if (location.state && location.state.email && !location.state.email.match(EMAIL_REGEX)) {
            return true;
        }

        if (captchaTypeValue === 'recaptcha' && !reCaptchaSuccess) {
            return true;
        }

        if (captchaTypeValue === 'geetest' && !geetestCaptchaSuccess) {
            return true;
        }

        return false;
    };
}

const mapStateToProps: MapStateToProps<ReduxProps, {}, RootState> = state => ({
    emailVerificationLoading: selectSendEmailVerificationLoading(state),
    i18n: selectCurrentLanguage(state),
    isMobileDevice: selectMobileDeviceState(state),
    error: selectSendEmailVerificationError(state),
    success: selectSendEmailVerificationSuccess(state),
    captcha_response: selectCaptchaResponse(state),
    reCaptchaSuccess: selectRecaptchaSuccess(state),
    geetestCaptchaSuccess: selectGeetestCaptchaSuccess(state),
    user: selectUserInfo(state),
});

const mapDispatchToProps = {
    emailVerificationFetch,
    resetCaptchaState,
};

export const EmailVerificationScreen = compose(
    injectIntl,
    withRouter,
    connect(mapStateToProps, mapDispatchToProps),
)(EmailVerificationComponent) as React.ComponentClass;
