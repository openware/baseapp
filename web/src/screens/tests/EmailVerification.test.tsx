import { render } from '@testing-library/react';
import React from 'react';
import { TestComponentWrapper } from 'src/lib/test';
import { IntlProps } from '../../';
import { EmailVerificationScreen } from '../EmailVerification';

const renderComponent = (props: Partial<IntlProps> = {}) =>
    render(
        <TestComponentWrapper>
            <EmailVerificationScreen />
        </TestComponentWrapper>,
    );

describe('EmailVerificationScreen', () => {
    it('should render', () => {
        expect(renderComponent().container).toMatchSnapshot();
    });
});
