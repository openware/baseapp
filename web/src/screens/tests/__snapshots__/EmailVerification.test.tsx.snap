// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailVerificationScreen should render 1`] = `
<div
  class="pg-emailverification-container"
>
  <div
    class="pg-emailverification"
  >
    <div
      class="pg-emailverification-title"
    >
      VERIFY YOUR EMAIL ADDRESS
    </div>
    <div
      class="pg-emailverification-body"
    >
      <div
        class="pg-emailverification-body-text"
      >
        To complete registration, check for an email in your inbox with further instruction. If you cannot find the email, please check your spam folder
      </div>
      <div
        class="pg-emailverification-body-container"
      >
        <button
          class="pg-emailverification-body-container-button"
        >
          Resend Confirmation
        </button>
      </div>
    </div>
  </div>
</div>
`;
